[release/2.0] Update go-cni version to fix Race Condition issue

MikeZappa87 opened the pull request:
Manual cherry-pick of https://github.com/containerd/containerd/pull/11244

Merging https://github.com/containerd/go-cni/pull/126 to fix race conditon in go-cni.

mikebrow reviewed (approved):
LGTM

mikebrow commented:
This is a manual cherry from main... auto cherry failed 3way diff (Auto-merging go.mod)

@MikeZappa87 can you edit the title and header to indicate this is a manual cherry pick?

samuelkarp commented:
I added `[release/2.0]` to the issue title and adjusted the description, but the commit messages differ between 09bf281ec415a6029177c60688e261dab55e3944 (on `main`) and a3104c8a25d6a0c69d8af56f5def41b0f94221ca in this PR.  @MikeZappa87, do you mind adjusting to match the first line and then indicate the source of the cherry-pick in a subsequent line of the commit message?

MikeZappa87 commented:
> I added `[release/2.0]` to the issue title and adjusted the description, but the commit messages differ between [09bf281](https://github.com/containerd/containerd/commit/09bf281ec415a6029177c60688e261dab55e3944) (on `main`) and [a3104c8](https://github.com/containerd/containerd/commit/a3104c8a25d6a0c69d8af56f5def41b0f94221ca) in this PR. @MikeZappa87, do you mind adjusting to match the first line and then indicate the source of the cherry-pick in a subsequent line of the commit message?

Done! Sorry just getting to this

MikeZappa87 commented:
/test pull-containerd-node-e2e

MikeZappa87 commented:
/test pull-containerd-node-e2e

MikeZappa87 commented:
/test pull-containerd-node-e2e

MikeZappa87 commented:
/test pull-containerd-node-e2e

MikeZappa87 commented:
@mikebrow looks like upstream fixed the issue and all tests are passing

MikeZappa87 commented:
/test pull-containerd-node-e2e

mikebrow reviewed (approved):
LGTM
